bentoml/archive/archiver.py

```diff
@@ -127,10 +134,53 @@ def _parse_requirements(file_path):
 
 cli=create_bentoml_cli(__module_path)
 
+
 def load():
     return archive.load(__module_path)
 
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-p', '--port')
+parser.add_argument('-w', '--workers')
+
+
+class GunicornApplication(gunicorn.app.base.BaseApplication):
+    def __init__(self, app, args=None):
+        if args:
+            parsed_args = parser.parse_args(args)
+            print(parsed_args)
+            self.options = {{}}
+            if parsed_args.workers:
+                self.options['workers'] = parsed_args.workers or (
+                    multiprocessing.cpu_count() // 2) + 1
+            if parsed_args.port:
+                self.options['bind'] = '%s:%s' % ('127.0.0.1', parsed_args.port) or '%s:%s' % (
+                    '127.0.0.1', '5000')
+        else:
+            self.options = {{
+                'workers': (multiprocessing.cpu_count() // 2) + 1,
+                'bind': '%s:%s' % ('127.0.0.1', '5000')
+            }}
+        self.application = app
+        super(GunicornApplication, self).__init__()
+
+    def load_config(self):
+        config = dict([(key, value) for key, value in iteritems(self.options)
+                       if key in self.cfg.settings and value is not None])
+        for key, value in iteritems(config):
+            self.cfg.set(key.lower(), value)
+
+    def load(self):
+        return self.application
+
+
 __all__ = ['__version__', '{service_name}', 'load']
+
+
+if __name__ == '__main__':
```

Comment:
On second thought, it might be better to put this inside a `bentoml` command. for example `bentoml serve-gunicorn ./bento_archive`. That way it's easier to re-use this code if we need to(as it won't be in a template), and a lot easier to test.